src/main/java/picard/sam/markduplicates/UmiAwareDuplicateSetIterator.java

```diff
@@ -128,23 +134,31 @@ private void process(final DuplicateSet set) {
 
         final UmiGraph umiGraph = new UmiGraph(set, umiTag, inferredUmiTag, allowMissingUmis);
 
-        List<DuplicateSet> duplicateSets = umiGraph.joinUmisIntoDuplicateSets(maxEditDistanceToJoin);
+        // Get the UMI metrics for the library of this duplicate set, creating a new one if necessary.
+        final String library = set.getRepresentative().getReadGroup().getLibrary();
+        UmiMetrics metrics = umiMetricsMap.get(library);
+        if (metrics == null) {
+            metrics = new UmiMetrics();
+            metrics.LIBRARY = library;
+            umiMetricsMap.put(library, metrics);
```

Comment:
I think you can simplify this with `computeIfAbsent`
```
        UmiMetrics metrics = umiMetricsMap.computeIfAbsent(library, key -> {
            UmiMetrics metrics1 = new UmiMetrics();
            metrics1.LIBRARY = key;
            return metrics1;
        });
```
If `UmiMetrics` had a constructor that accepts `library` as an argument it would be even simpler:
```
UmiMetrics metrics = umiMetricsMap.computeIfAbsent(library, UmiMetrics::new);
```